internal/policy/filter.go

```diff
@@ -0,0 +1,70 @@
+/*
```

Comment:
Not super happy about how this was laid out, couldn't think of a nicer way to implement it at the moment though.